lightning/src/blinded_path/message.rs

```diff
@@ -138,17 +136,25 @@ pub(super) fn blinded_hops<T: secp256k1::Signing + secp256k1::Verification>(
 ) -> Result<Vec<BlindedHop>, secp256k1::Error> {
 	let pks = intermediate_nodes.iter().map(|node| &node.node_id)
 		.chain(core::iter::once(&recipient_node_id));
-	let tlvs = pks.clone()
+	let tlvs: Vec<ControlTlvs> = pks.clone()
 		.skip(1) // The first node's TLVs contains the next node's pubkey
 		.zip(intermediate_nodes.iter().map(|node| node.short_channel_id))
 		.map(|(pubkey, scid)| match scid {
 			Some(scid) => NextMessageHop::ShortChannelId(scid),
 			None => NextMessageHop::NodeId(*pubkey),
 		})
 		.map(|next_hop| ControlTlvs::Forward(ForwardTlvs { next_hop, next_blinding_override: None }))
-		.chain(core::iter::once(ControlTlvs::Receive(ReceiveTlvs{ context: Some(context) })));
+		.chain(core::iter::once(ControlTlvs::Receive(ReceiveTlvs{ context: Some(context) })))
+		.collect();
 
-	utils::construct_blinded_hops(secp_ctx, pks, tlvs, session_priv)
+	let max_length = tlvs.iter()
+		.max_by_key(|c| c.serialized_length())
+		.map(|c| c.serialized_length())
+		.unwrap_or(0);
+
+	let length_tlvs = tlvs.into_iter().map(move |tlv| (max_length, tlv));
+
+	utils::construct_blinded_hops(secp_ctx, pks, length_tlvs, session_priv)
```

Comment:
Could you add a test that checks if `BlindedHop::encrypted_payload` in each hop of a `BlindedPath` has the same length?

Reply:
Done! [pr3177.02](https://github.com/shaavan/rust-lightning/commits/pr3177.02)

Introduce a test, and a test util to make sure encrypted payloads are of the same size for blinded message and payment paths.

Thanks a lot for the pointer!